modules/es7.promise.finally.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+// https://github.com/tc39/proposal-promise-finally
+var $export = require('./_export');
+
+$export($export.P, 'Promise', {
+  'finally': function _finally(onFinally){
+    var handler = typeof onFinally === 'function' ? onFinally : function(){};
+    var C = Promise;
+    var newPromise = Promise.prototype.then.call(
+      this, // throw if IsPromise(this) is not true
+      function(x) {
+        return new C()(function(resolve) {
```

Comment:
@kovensky:

1. Thanks for the older implementation reference. That's what I would expect. I like to move methodically through uncharted waters.
1. I'll use the core-js internal implementation of `speciesConstructor` used in `es6.promise.js`
1. Thanks for the quick responses!